gateway/gateway_test.go

```diff
@@ -341,16 +345,26 @@ var _ = Describe("Gateway", func() {
 			gateway    *HandleT
 			statsStore *memstats.Store
 			whServer   *httptest.Server
+			serverURL  string
 		)
 
 		BeforeEach(func() {
 			whServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 				w.WriteHeader(http.StatusOK)
 				_, _ = io.WriteString(w, "OK")
 			}))
-			url, err := url.Parse(whServer.URL)
+			WHURL := whServer.URL
+			parsedURL, err := url.Parse(WHURL)
+			Expect(err).To(BeNil())
+			whPort := parsedURL.Port()
+			os.Setenv("RSERVER_WAREHOUSE_WEB_PORT", whPort)
```

Comment:
It should be possible to use `GinkgoT().Setenv()`. Just check it out, if not nevermind.